src/ranged.cpp

```diff
@@ -1536,14 +1536,34 @@ void drop_or_embed_projectile( const dealt_projectile_attack &attack )
     item dropped_item = drop_item;
 
     monster *mon = dynamic_cast<monster *>( attack.hit_critter );
-    // Try to embed the projectile in monster
-    // Don't embed on miss, in player/NPCs, when we didn't stab/cut properly
-    //  or when the item simply shouldn't be embedded (for example, it is active)
-    if( mon == nullptr || mon->is_dead_state() ||
-        ( attack.dealt_dam.type_damage( DT_STAB ) +
-          attack.dealt_dam.type_damage( DT_CUT ) <=
-            attack.dealt_dam.type_damage( DT_BASH ) ) ||
-        effects.count( "NO_EMBED" ) != 0 ) {
+
+    // We can only embed in monsters
+    bool embed = mon != nullptr && !mon->is_dead_state();
+    // And if we actually want to embed
+    embed = embed && effects.count( "NO_EMBED" ) == 0;
+    // Don't embed in small creatures
+    if( embed ) {
+        const m_size critter_size = mon->get_size();
+        const int vol = dropped_item.volume( true, false );
```

Comment:
Yes, but you already have a single item (and if it's counted by charges, it will already have 1 charges only, see end of `game::plthrow`). I have to agree here, `charges` will be 1 and `ret *= charges` will not do anything.

But it's a more formal matter, it's confusing because in theory, it could return less volume explicitly calling with `true` is not needed. It makes the impression that the item *may* have more charges, which btw. would be added to the monster and later dropped anyway.

So in short: if I somehow manage to throw a stack of 100 throwing knifes (should not be possible, but who knows), this function will use the volume of 1 knife, but will add 100 knifes to the monster inventory.